lib/ember-test-helpers/test-module.js

```diff
@@ -269,6 +271,36 @@ export default Klass.extend({
     };
   },
 
+  /*
+    Builds a version of the passed in context that contains deprecation warnings
+    for accessing properties that exist on the module.
+  */
+  _buildDeprecatedContext: function(module, context) {
+    var deprecatedContext = Object.create(context);
+
+    var keysForDeprecation = Object.keys(module);
+
+    for (var i = 0, l = keysForDeprecation.length; i < l; i++) {
+      this._proxyDeprecation(module, deprecatedContext, keysForDeprecation[i]);
+    }
+
+    return deprecatedContext;
+  },
+
+  /*
+    Defines a key on an object to act as a proxy for deprecating the original.
+  */
+  _proxyDeprecation: function(obj, proxy, key) {
+    if (typeof proxy[key] === 'undefined') {
+      Object.defineProperty(proxy, key, {
+        get: function() {
+          Ember.deprecate('Accessing the test module property "' + key + '" from a callback is deprecated.', false, { id: 'ember-test-helpers.test-module.callback-context', until: '0.6.0' });
```

Comment:
The `until` value here is tricky.  I agree that 0.6.0 is the correct version number of this project, but none of this projects consumers actually see this version.  For example, ember-mocha is already on 0.8.  

It is likely that nothing can be done here, but I wanted to call it out and see if y'all had any ideas.

/cc @mixonic @dgeb